core/src/main/java/org/hisp/dhis/android/core/arch/db/adapters/custom/internal/DbVisualizationLegendColumnAdapter.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (c) 2004-2022, University of Oslo
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  Redistributions of source code must retain the above copyright notice, this
+ *  list of conditions and the following disclaimer.
+ *
+ *  Redistributions in binary form must reproduce the above copyright notice,
+ *  this list of conditions and the following disclaimer in the documentation
+ *  and/or other materials provided with the distribution.
+ *  Neither the name of the HISP project nor the names of its contributors may
+ *  be used to endorse or promote products derived from this software without
+ *  specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ *  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.android.core.arch.db.adapters.custom.internal
+
+import android.content.ContentValues
+import android.database.Cursor
+import com.gabrielittner.auto.value.cursor.ColumnTypeAdapter
+import org.hisp.dhis.android.core.visualization.VisualizationLegend
+import org.hisp.dhis.android.core.visualization.VisualizationTableInfo
+
+class DbVisualizationLegendColumnAdapter : ColumnTypeAdapter<VisualizationLegend> {
+
+    override fun fromCursor(cursor: Cursor, columnName: String): VisualizationLegend? {
+        val legendShowKeyColumnIndex = cursor.getColumnIndex(VisualizationTableInfo.Columns.LEGEND_SHOW_KEY)
+        val legendStrategyColumnIndex = cursor.getColumnIndex(VisualizationTableInfo.Columns.LEGEND_STRATEGY)
+        val legendStyleColumnIndex = cursor.getColumnIndex(VisualizationTableInfo.Columns.LEGEND_STYLE)
+
+        val legendShowKey = cursor.getString(legendShowKeyColumnIndex)
+        val legendStrategy = cursor.getString(legendStrategyColumnIndex)
+        val legendStyle = cursor.getString(legendStyleColumnIndex)
+
+        val isLegendInfoPresent = listOf(legendShowKey, legendStrategy, legendStyle).any {
+            it != null
+        }
+
+        return if (isLegendInfoPresent) {
```

Comment:
Here you have to build the VisualizationLegend object using all the properties, not just legendShowKey. The condition to create is right (if any of them is not null, the object must be created), but the object must contain all the proporties, even if some of them are null.